File: web/types/project.ts
import type { ProjectStatus } from "./enums";
import type { TagResponse } from "./tag";

export type ProjectResponse = {
  id: number;
  userId: number;
  name: string;
  description: string | null;
  goal: string | null;
  status: ProjectStatus;
  repoUrl: string | null;
  lastActivityAt: string | null;
  completedAt: string | null;
  createdAt: string;
  updatedAt: string | null;
  isDeleted: boolean;
  deletedAt: string | null;
  tags: TagResponse[];
};

export type ProjectCreateRequest = {
  name: string;
  description?: string | null;
  goal?: string | null;
  repoUrl?: string | null;
  status?: ProjectStatus;
};

export type ProjectUpdateRequest = {
  name: string;
  description?: string | null;
  goal?: string | null;
  repoUrl?: string | null;
};

export type ProjectStatusUpdateRequest = {
  status: ProjectStatus;
};

export type ProjectListQuery = {
  status?: ProjectStatus;
  tagId?: number;
  page?: number;
  pageSize?: number;
  includeDeleted?: boolean;
};

File: src/DevTrack.Domain/Entities/Project.cs
using DevTrack.Domain.Enums;

namespace DevTrack.Domain.Entities;

public class Project : SoftDeletableEntity
{
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Goal { get; set; }
    public ProjectStatus Status { get; set; } = ProjectStatus.Active;
    public string? RepoUrl { get; set; }
    public DateTime? LastActivityAt { get; set; }
    public DateTime? CompletedAt { get; set; }

    public User User { get; set; } = null!;
    public ICollection<Component> Components { get; set; } = new List<Component>();
    public ICollection<ProjectTag> ProjectTags { get; set; } = new List<ProjectTag>();
}
